slobrok/src/vespa/slobrok/server/local_rpc_monitor_map.cpp

```diff
@@ -120,14 +121,14 @@ void LocalRpcMonitorMap::notifyFailedRpcSrv(ManagedRpcServer *rpcsrv, std::strin
         auto target = std::move(psd.inflight);
         target->doneHandler(OkState(13, "failed check using listNames callback"));
     }
-    if (psd.up) {
-        psd.up = false;
-        _dispatcher.remove(psd.mapping());
-    }
     if (psd.localOnly) {
         auto iter = _map.find(psd.name());
         _map.erase(iter);
     }
+    if (psd.up) {
```

Comment:
should not assume that we are not deleted by the erase above (even if it probably will be impossible)